src/db.c

```diff
@@ -664,6 +664,7 @@ long long emptyDbStructure(redisDb *dbarray, int dbnum, int async,
             dbarray[j].sub_dict[subdict].key_count = 0;
             dbarray[j].sub_dict[subdict].resize_cursor = 0;
             if (server.cluster_enabled) {
+                dbarray[j].sub_dict[subdict].bucket_count = 0;
```

Comment:
Question that suddenly occurred to me, do we need to empty the rehashing list?

Reply:
Yes, according to the current rehashing method, it's better clear the list to avoid those meaningless attempts in incrementallyRehash. However, TBH, I am not a fan of the current approach of using pointer list for rehashing. I'm thinking if there could be a better way to completely eliminate the need for rehashing lists.

Reply:
oops, i just saw it, i just submitted this PR #12764.
yean, i agree, now the rehashing list, doesn't seem to like it that much.